### PR TITLE
chore(topology): split `build_pieces` into smaller functions

### PR DESCRIPTION
**Pull request by StephenWakely**
A tiny refactor, this function was getting a bit too long.

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]

File: src/topology/builder.rs [hunk 1]
```diff
@@ -157,6 +156,78 @@ pub async fn build_pieces(
...
+        config,
+        diff,
+        &mut errors,
+        buffers,
+        &mut inputs,
+        &mut healthchecks,
+        &mut tasks,
+        &mut detach_triggers,
+    )
+    .await;
```
Comment (bruceg) [inline src/topology/builder.rs:190]: I'd like to see these repeated parameters moved into a common struct or something, and then the functions could be member functions of that struct. Notable, all of the following are repeated: `config`, `diff`, `errors`, `inputs`, `outputs`, and `tasks`.

File: src/topology/builder.rs [hunk 2]
```diff
@@ -157,6 +156,78 @@ pub async fn build_pieces(
...
+            inputs,
+            outputs: finalized_outputs,
+            tasks,
+            source_tasks,
+            healthchecks,
+            shutdown_coordinator,
+            detach_triggers,
+        };
+
+        Ok(pieces)
```
Comment (bruceg) [inline src/topology/builder.rs:215]: nit: shorter:
```suggestion
        Ok(Pieces {
            inputs,
            outputs: finalized_outputs,
            tasks,
            source_tasks,
            healthchecks,
            shutdown_coordinator,
            detach_triggers,
        })
```

**Review by bruceg — COMMENTED** [review]
LGTM except for the long argument lists, which wouldn't be a big deal except that most of them are repeated.

File: src/topology/builder.rs (hunk 1 above)
Comment (StephenWakely) [reply src/topology/builder.rs]: I really like this. It has made the diffs for the PR much harder to read, but it is largely just moving code around..

**Pull request merged**